packages/hdwallet-provider/src/index.ts

```diff
@@ -155,10 +148,27 @@ class HDWalletProvider {
           } else {
             cb("Account not found");
           }
-          const chain = (chainId && typeof chainId === "string") ?
-            parseInt(chainId) :
-            chainId;
-          const tx = new Transaction(txParams, { chain });
+          const chain = (self.chainId && typeof self.chainId === "string") ?
+          // @ts-ignore
+            parseInt(self.chainId) :
```

Comment:
I think you should parse the `chainId` to a number up front, where it is set.

Reply:
Yeah this is kind of a leftover from earlier. Currently I have this typed as a number, is it worth allowing it as a string as well?